nativescript-3.1.0-timeline/README.md

```diff
@@ -0,0 +1,236 @@
+'[Using Webpack to Bundle Your Code](https://docs.nativescript.org/best-practices/bundling-with-webpack#inspecting-bundles)' NativeScript article.
+
+In general you should move everything from node_modules to the vendor. However some plugins that use native Android APIs will have to be refactored as the native APIs are not exposed during the snapshot generation. You can check the PR for the [nativescript-plugin-firebase made by Stanimira Vlaeva](https://github.com/EddyVerbruggen/nativescript-plugin-firebase/pull/424), it wraps the access to Android Java classes in functions and delay their execution until runtime. When it is merged, a `require("nativescript-plugin-firebase")` should also be added to the `vendor.ts`.
+
+Now this is the transition from 3.0.0 to 3.1.0 and snapshot explained:
+![timeline-view-screenshot-explained](./timelines/how-it-feels.png)
+
+### iOS
+#### Timeline: tns run ios
+Running vanilla ios build:
+```bash
+tns run ios | timeline-view
+```
+
+[iOS vanilla run HTML times report](./timelines/tns-run-ios.html).
+
+The app does:
+ - 400ms. preparing for live-sync
+ - 500ms. waiting for debugger
+ - 900ms. require nativescript-angular/platform
+ - 850ms. require app.module.js
+    - 500ms. require rxjs/Rx.js (this time is already included as part of the app.module.js)
+ - ~1000ms making views
+
+A total of 3sec. (Don't do the math yet)
+
+This however shows about 900ms are spent in the main method that are executed only in debug. The waiting for debug time is removed in release. Extracting from the bundle is also removed, as it is part of the live-sync logic, and live-sync is not available in release. So release build would rather take 3sec.
+
+#### Timeline: npm run start-ios-bundle --uglify
+Running the iOS application with webpack, the Angular 2 tooling and uglify gets the maximum out of an iOS application.
+```bash
+npm run start-ios-bundle --uglify | timeline-view
+```
+
+[iOS bundle and uglify HTML times report](./timelines/npm-run-ios-bundle-uglify.html).
+
+The app does:
+ - 500ms. waiting for debugger
+ - 400ms. in vendor.js
+ - ~500ms. making views
+
+That is a little under a sec.
+
+It is pretty plain and boring compared to the previous and the android versions. Bundling packs all files in a single file reducing IO, and the uglification has much higher impact since JavaScriptCore on iOS does no JIT compilation. Subtracting the 500ms. debug. The bundle extraction happens again but it is fast, in the previous case it was slow because of the number of files.
+
+## Sweet Awesomeness
+With fast times comes great responsibility. Now that your app starts under a second on iOS and somewhat acceptable two seconds on Android, how do we make the best impression for the transition from the SpringBoard to your app's content? You can't put your app's slogan nor a fancy animation since there is simply not enough time for the user to consume it.
```

Comment:
I'll remove the "somewhat". 👍 2sec. is the baseline and is "somewhat acceptable" Imo :) but we made it :) when we remove the GC and the asset extraction it would be perfect!